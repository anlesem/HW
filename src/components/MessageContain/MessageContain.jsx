import { useCallback } from 'react';
import { useSelector, useDispatch } from 'react-redux';

import { getMessageList } from '../../store/messages/selectors';
import { addMessageThunk } from '../../store/messages/actions';

import { MessageList } from '../MessageList/MessageList';
import { MessageForm } from '../MessageForm/MessageForm';

import style from './MessageContain.module.scss';

export const MessageContain = ({ chatId }) => {
  const dispatch = useDispatch();
  const messages = useSelector(getMessageList);

  const sendMassage = useCallback(
    (text, author, id) => {
      dispatch(addMessageThunk({ chatId, text, author, id }));
    },
    [chatId, dispatch]
  );

  return (
    <div className={style.wrap}>
      <MessageList message={messages[`chat${chatId}`]} />
      <MessageForm formData={{ chatId, sendMassage }} />
    </div>
  );
};
