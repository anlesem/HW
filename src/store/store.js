import { createStore, combineReducers, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';

import { profileReducer } from './profile/reducer';
import { chatsReducer } from './chats/reducer';
import { messagesReducer } from './messages/reducer';
import { dataReducer } from './data/reducer';

const persistConfig = {
  key: 'root',
  storage,
  blacklist: ['data']
};

const rootReducers = combineReducers({
  profile: profileReducer,
  chats: chatsReducer,
  messages: messagesReducer,
  data: dataReducer
});

const persistedReducers = persistReducer(persistConfig, rootReducers);

export const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const store = createStore(persistedReducers, composeEnhancers(applyMiddleware(thunk)));

export const persistor = persistStore(store);
