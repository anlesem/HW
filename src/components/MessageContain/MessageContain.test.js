import { render, screen } from '@testing-library/react';
import { MessageContain } from './MessageContain';

import { Provider } from 'react-redux';
import { store } from '../../store/store';

describe('MessageContain', () => {
  it('Компонент существует', () => {
    expect(MessageContain).toBeInstanceOf(Function);
  });

  it('Снимок состояния', () => {
    const chatId = 1;
    const { asFragment } = render(
      <Provider store={store}>
        <MessageContain chatId={chatId} />
      </Provider>
    );
    expect(asFragment(<MessageContain />)).toMatchSnapshot();
  });

  it('Отображает приветственное сообщение', () => {
    const chatId = 1;
    render(
      <Provider store={store}>
        <MessageContain chatId={chatId} />
      </Provider>
    );

    expect(screen.getByText(/добро пожаловать/i)).toBeTruthy();
  });
});
