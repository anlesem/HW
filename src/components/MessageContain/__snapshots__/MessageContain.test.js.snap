// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageContain Снимок состояния 1`] = `
<DocumentFragment>
  <div
    class="wrap"
  />
</DocumentFragment>
`;
